chain/client/src/chunk_validation.rs

```diff
@@ -525,13 +534,42 @@ impl Client {
     }
 
     /// Function to process an incoming chunk endorsement from chunk validators.
+    /// We first verify the chunk endorsement and then store it in a cache.
+    /// We would later include the endorsements in the block production.
     pub fn process_chunk_endorsement(
         &mut self,
-        _endorsement: ChunkEndorsement,
+        endorsement: ChunkEndorsement,
     ) -> Result<(), Error> {
-        // TODO(10265): Here if we are the current block producer, we would store the chunk endorsement
-        // for each chunk which would later be used during block production to check whether to include the
-        // chunk or not.
+        let chunk_hash = endorsement.chunk_hash();
```

Comment:
Should we also add a check if a node is a block producer for the height?

Reply:
Thought of this one as well, this would definitely be a good thing to have from security/adversarial point of view. For a general case, we do have code that directs the endorsement to only the block producer tho.
```
    let endorsement = ChunkEndorsement::new(chunk_header.chunk_hash(), signer);
    network_sender.send(PeerManagerMessageRequest::NetworkRequests(
        NetworkRequests::ChunkEndorsement(block_producer, endorsement),
    ));
```

Let me know if you feel it's a simple enough check to add.

Reply:
Let me try to figure out a way...

Reply:
Aah crap, this raises a broader question in my head wrt chunk endorsements in case when chunks are missing in blocks... Will follow up offline